### PR TITLE
*: use async get timestamp API && remove asyncGetTSWorker

### PR DESCRIPTION
**Pull request by tiancaiamao**
With the new GetTSAsync API, a session doesn't need to bind a `asyncGetTSWorker` goroutine any more.

This change brings those benefits:
1. Eliminate potential goroutine leak when `session.Close` is not called.
2. Good for golang runtime scheduler as it eliminate a mass of background goroutines.

### CODE REVIEW

File: session.go [hunk 1]
```diff
@@ -1059,30 +1056,24 @@ func (s *session) loadCommonGlobalVariablesIfNeeded() error {
...
+	store  kv.Storage
 }
 
-func (s *session) getTxnFuture() *txnFuture {
-	if s.txnFutureCh == nil {
-		s.txnFutureCh = make(chan *txnFuture, 1)
-		go asyncGetTSWorker(s.store, s.txnFutureCh)
+func (tf *txnFuture) wait() (kv.Transaction, error) {
+	startTS, err := tf.future.Wait()
+	if err == nil {
```
Comment (shenli) [inline session.go]: How to handle the error?

File: session.go [hunk 2]
```diff
@@ -1095,9 +1086,8 @@ func (s *session) prepareTxnCtx() {
 		return
 	}
 
-	txnFuture := s.getTxnFuture()
-	goCtx, cancelFunc := goctx.WithCancel(goctx.Background())
-	s.txnFuture, s.goCtx, s.cancelFunc = txnFuture, goCtx, cancelFunc
+	s.goCtx, s.cancelFunc = goctx.WithCancel(goctx.Background())
```
Comment (zimulala) [inline session.go]: Why store the Context inside a struct type?

**Comment by coocood** [comment]
LGTM

File: session.go (hunk 1 above)
Comment (tiancaiamao) [inline session.go]: This error is always `Canceled`, caused by network timeout.
The real reason may be pd leader change or server down or network problem...

But it doesn't matter, if err not nil, we'll call `store.Begin()` in line 1070, 
which would retry get timestamp until timeout.

File: session.go (hunk 2 above)
Comment (tiancaiamao) [inline session.go]: For historical reasons.
Each transaction goes along with a new context, transaction run in a session,
so the context is stored in the session.

**Comment by shenli** [comment]
LGTM

**Pull request merged**